### PR TITLE
Updating the Cazyme Annotation Pipeline

### PR DESCRIPTION
**Pull request by miraep8**
Main changes:

- Rather than taking in the raw metagenomic reads, now `run_dbcan` takes in the predicted genes from `metaerg`.  This has the advantage that the annotated CAZyme genes will match genes annotated using other tools. 
- R1/R2  are aligned to the genes predicted using metaerg.  I think this is useful if you want to any sort of annotation-related analysis that goes beyond presence/absence analysis. 
- deleting `aggregate_cazi_results` and `aggregate_metaerg_results` functions.  My impression is that these aren't used anywhere, so removing them to reduce confusion. (At first I thought these were defining the inputs for the joining functions, but I don't think it is actually the case). 

This works as is now, however I think there are a few things to improve on pre-merging.  Will probably just use a combination bwa/samtools docker container rather than having to split the alignment and sorting/compression into different jobs. Also I plan to do aligning on the fragmented meta-erg output rather than the joined files which should speed up that component considerably (currently a bit on the slow side).

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -90,13 +90,14 @@ docker_abricate: "docker://staphb/abricate:1.0.0"
...
 docker_bbtools: "docker://staphb/bbtools:38.97"
+docker_bedtools: "docker://staphb/bedtools:2.31.1"
 docker_biobakery: "docker://ghcr.io/vdblab/biobakery-profiler:20230418a"
 docker_biopython: "docker://nanozoo/biopython:1.74--b03961c"
 docker_bowtie2: "docker://ghcr.io/vdblab/bowtie2:2.5.0"
 docker_bmtagger: "docker://ghcr.io/vdblab/kneaddata-bmtagger:0.6.1"
 docker_coverm: "docker://ghcr.io/vdblab/coverm:0.6.1"
 docker_cutadapt: "docker://ghcr.io/vdblab/cutadapt:3.7"
-docker_dbcan: "docker://haidyi/run_dbcan:3.0.1"
+docker_dbcan: "docker://baichom1/dbcan3:4.1.4.12"
```
Comment (nickp60) [inline config/config.yaml:100]: Does this image differ from the one in the dockerfiles repo?

File: workflow/rules/annotate.smk [hunk 2]
```diff
@@ -242,76 +248,136 @@ rule annotate_CAZI_split:
...
         """
         # deal with header
-        head -n 1 {input.gff[0]} > {output}
-        for f in {input}
+        head -n 1 {input.gff[0]} > {output.gff}
+        for f in {input.gff}
+        do
+            IFS='/'; name_arr=($f); unset IFS;
+            part=${{name_arr[@]: -3: 1}}
+            tail -n+2 $f >> seqkit replace -p 'ID=[^;]*;' -r 'ID=megahit_'$part'_{{nr}};' $f >> {output.gff}
```
Comment (nickp60) [inline workflow/rules/annotate.smk:301]: Would this need changes to be compatible with spades assemblies?

**Review by nickp60 — COMMENTED** [review]
Looks good overall!  Do you think it would be worth including any of the outputs from the dbcan_utils scripts to get RPKM type outputs?

File: workflow/rules/annotate.smk (hunk 2 above)
Comment (miraep8) [reply workflow/rules/annotate.smk:301]: Calling it `megahit` here shouldn't be important to anything downstream, but will call it something else just to be on the safe side. 👍 Will test it with some of the spades output as well just to make sure.

**Pull request merged**